backend/tournesol/views/unconnected_entities.py

```diff
@@ -0,0 +1,81 @@
+"""
+API endpoints to show unconnected entities
+"""
+from collections import defaultdict
+
+from django.db.models import ObjectDoesNotExist
+from django.http import Http404
+from drf_spectacular.utils import extend_schema, extend_schema_view
+from rest_framework import generics
+from rest_framework.permissions import IsAuthenticated
+
+from tournesol.views.mixins.poll import PollScopedViewMixin
+
+from ..models import Comparison, Entity
+from ..serializers.entity import EntitySerializer
+
+
+@extend_schema_view(
+    get=extend_schema(description="Show unconnected entities for the current user")
+)
+class UnconnectedEntitiesView(
+    PollScopedViewMixin,
+    generics.ListAPIView
+):
+    """
+    API view for showing unconnected entities
+    """
+    serializer_class = EntitySerializer
+    permission_classes = [IsAuthenticated]
+
+    __already_visited_node__ = set()
+    __all_entites__ = set()
+    __all_connections__ = defaultdict(set)
+
+    def get_related_entities(self, entity_id):
+
+        self.__already_visited_node__.add(entity_id)
+        self.__all_entites__.remove(entity_id)
```

Comment:
This line raises a 500 error in the case where a user calls the endpoint with the uid of an Entity that exists but they have not yet compared.

I think it would be easily solved and the appropriate behaviour would be to return `__all_entites__` in the case of a not compared entity uid ?

I would recommend adding this in the test suite too :)

Reply:
Yes good catch. I switch the purpose of the `__all_entities__` as `_entities_without_comparison_` to fix this issue (I added a test case). Every entities without comparison with the same type are now also return from the API